### PR TITLE
Correcting duplicates

### PR DESCRIPTION
**Pull request by aammd**
Here i've tried to resolve the duplicate taxa in the most conservative way. there were three approaches:

* differ only in `key` ( "simple" duplicates) -- deleted
* two "different taxa" with the same nickname -- simply made a new nickname.
* difference in `name1`, `name2` -- added synonyms

### CODE REVIEW

**Comment by aammd** [comment]
cc @adletaw , would you take a look at this? though I was thinking that this would be a good opportunity to demonstrate how we would get changes reviewed by Diane.

File: Rscripts/12_correcting_duplicates.R [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+## based on report.Rmd
+
+## finding doubled rows
+
+#' Find duplicate nicknames
+#'
+#' @param DF dataframe to check for duplicate nicknames
+#'
+#' @return a character vector of the nicknames of duplicates.
+find_duplicates <- function(DF){
```
Comment (adletaw) [inline Rscripts/12_correcting_duplicates.R]: Add this function to `functions.R` ?

File: Rscripts/12_correcting_duplicates.R [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  filter(nickname %in% find_duplicates(df)) %>% 
+  daff_duplicates()
+
+
+#' Which columns differ between duplicates
+#'
+#' @param DF data.frame containing duplicate nicknames
+#'
+#' @return named list. names are duplicate nicknames, elements are the columns that vary
+which_cols_different <- function(DF){
```
Comment (adletaw) [inline Rscripts/12_correcting_duplicates.R]: This is awesome.

File: Rscripts/12_correcting_duplicates.R [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+df_nosimple %>% 
+  filter(nickname %in% find_duplicates(DF = df_nosimple)) %>% 
+  daff_duplicates()
+
+### Tricoptera.5602 is Trichoptera.2 ----
+
+df_nosimple_trich <- df_nosimple
+
+df_nosimple_trich[which(df_nosimple_trich$Name %in% "Trichoptera.5602"), "nickname"] <- "Trichoptera.2"
+
```
Comment (adletaw) [inline Rscripts/12_correcting_duplicates.R]: I think you can use the `reclassify` function to rename this Trichoptera

File: Rscripts/12_correcting_duplicates.R [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+## whats the max beetle?!
+find_max_nick <- function(Taxa, DF){
+  str_detect %>% 
+    partial(pattern = Taxa) %>% 
+    Filter(DF$nickname) %>% 
+    str_replace_all("[A-Za-z.]", "") %>% 
+    as.numeric %>% 
+    max
+}
```
Comment (adletaw) [inline Rscripts/12_correcting_duplicates.R]: This `find_max_nick` function could be really useful in the future.. :+1:

File: Rscripts/12_correcting_duplicates.R [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+
+# Microculex stonei != Unknown Diptera L --------------
+
+
+find_max_nick("Diptera", DF = df_nosimple_trich)
+# 534
+
+df_nosimp_trich_hydro_unk <- df_nosimp_trich_hydro
+
+df_nosimp_trich_hydro_unk[which(df_nosimp_trich_hydro_unk$Name %in%
```
Comment (adletaw) [inline Rscripts/12_correcting_duplicates.R]: `reclassify` again.

**Comment by adletaw** [comment]
This looks great to me!  It would be nice to put the functions into our `functions.R` file.. it would make it easier to read.. unless you have different thoughts on the subject?  There are also 2 or 3 places you can use `reclassify` to change the nickname of a species.

**Pull request merged**